### PR TITLE
Small performance changes on TextInputModal + new "allowEmpty" var + fixes Cargo's weird TGUI pop-up

### LINKED ISSUE #11903: When you order something through Cargo, only the Cancel button works.
### Labels

[BUG]

### Description

When you order something in the "Place Order" tab, the pop-up asking for any special labbeling has a "Cancel" and "OK" button, but the OK button is disabled, and you can order something anyway with the Cancel button.

### Steps to reproduce

1. Go to a QM console.
2. Try to order anything.
3. Why did they TGUI almost every in-game prompt anyway?

### Expected Behavior

For the OK button to be enabled.

### Additional Information & Screenshots

And optionally make the Cancel button actually cancel the order.

### PR DESCRIPTION
**Pull request by CaioFrancisco**
## About the PR 
The small performance changes on TextInputModal are just moving the static vars to `ui_static_data` and applying the new `allowEmpty` var on the initial `inputIsValid` var, fixing the Submit button blinking green for a split second after opening the TextInputModal.

The new "allowEmpty" var on TextInputModal pop-ups allow for the "Submit" button to be available even when the input box is empty (returning ""). I've gone ahead and applied it to some of the pre-existing `tgui_input_text`s.

The weird Cargo pop-up is how independent of what you do to the pop-up, it will still order whatever you clicked on. This fixes that, and allows for crates to be ordered without forcing the user to write something to actually order a crate.

## Why's this needed? 
Fixes are good. Also, tell me whether I should atomize this PR, although I don't see the need of it since it's just bugfixes + an internal feature.
Fixes #11903

### CODE REVIEW

**Comment by CaioFrancisco** [comment]
That backfired. I'll fix it and set it back ready to review.

**Comment by CaioFrancisco** [comment]
I was actually lucky to have it not compile, or I would have missed a glaring bug: the money was deducted from Cargo before the special tag was prompted.

File: code/obj/machinery/computer/QM_supply.dm [hunk 1]
```diff
@@ -530,11 +530,14 @@ var/global/datum/cdc_contact_controller/QM_CDC = new()
...
 					shippingmarket.supply_requests -= O
 					if(wagesystem.shipping_budget >= P.cost)
-						wagesystem.shipping_budget -= P.cost
 						O.object = P
 						O.orderedby = usr.name
 						var/default_comment = ""
-						O.comment = copytext(html_encode(tgui_input_text(usr, "Comment:", "Enter comment", default_comment, multiline = TRUE)), 1, MAX_MESSAGE_LEN)
+						O.comment = tgui_input_text(usr, "Comment:", "Enter comment", default_comment, multiline = TRUE, max_length = 32, allowEmpty = TRUE)
+						if (O.comment == null)
+							return .("list")// The user cancelled the order
```
Comment (ZeWaka) [inline code/obj/machinery/computer/QM_supply.dm:538]: this is so jank

**Review by ZeWaka — APPROVED** [review]

File: code/obj/machinery/computer/QM_supply.dm (hunk 1 above)
Comment (CaioFrancisco) [reply code/obj/machinery/computer/QM_supply.dm:538]: I'll try to ease the jank with the power of macros.

Comment (CaioFrancisco) [reply code/obj/machinery/computer/QM_supply.dm]: There we go. Tell me if it's worse or better.

Comment (ZeWaka) [reply code/obj/machinery/computer/QM_supply.dm]: oh i mean `.("list")` lol

Comment (ZeWaka) [reply code/obj/machinery/computer/QM_supply.dm]: but yes those changes r good

**Comment by thegrb93** [comment]
@CaioFrancisco could you look into the linked issue? I think this is gray-ing out the submit button despite the text entry getting default text.

**Comment by thegrb93** [comment]
The Chemaster is where I first noticed the issue happening

**Pull request merged**